src/App.tsx

```diff
@@ -1,223 +1,153 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+  useCallback,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { getTodos, addTodo, deleteTodo } from './api/todos';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { filteredTodosFunction } from './components/Helpers/FilteredTodos';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState(Filter.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isNewTodoLoading, setIsNewTodoLoading] = useState(false);
+
   useEffect(() => {
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setErrorMessage('Failed to load todos');
+        });
+    }
+  }, []);
+
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed).length
+  ), [todos]);
+
+  const comletedTodos = useMemo(() => (
+    todos.some(todo => todo.completed)
+  ), [todos]);
+
+  const filteredTodos = useMemo(() => (
+    filteredTodosFunction(todos, status)
+  ), [todos, status]);
+
+  const createTodo = useCallback(
+    async (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      try {
+        if (!title.trim()) {
+          setErrorMessage('Title can\'t be empty');
```

Comment:
Good practice to use ENUM for errors, so please fix it in the next task
```suggestion
          setErrorMessage(TodoErrors.EmptyTitle);
```

Reply:
Sure, thank you!